packages/main/src/themes/TextArea.css

```diff
@@ -166,158 +259,155 @@
 }
 
-:host(:not([value-state]):not([readonly]):hover) .ui5-textarea-inner {
-	background-color: var(--sapField_Hover_Background);
+:host([value-state="Error"][focused]:not([readonly])) .ui5-textarea-wrapper:after {
+	border-color: var(--_ui5_textarea_focused_value_state_error_focus_outline_color);
 }
 
-:host([value-state]:not([value-state="None"])) .ui5-textarea-inner {
-	border-width: var(--_ui5_textarea_state_border_width);
+:host([value-state="Error"]:not([readonly]):not([focused]):hover) .ui5-textarea-wrapper {
+	background-color: var(--_ui5_textarea_error_hover_background_color);
+	box-shadow: var(--_ui5_textarea_value_state_error_hover_box_shadow);
 }
 
 :host([value-state="Error"]) .ui5-textarea-inner,
 :host([value-state="Warning"]) .ui5-textarea-inner {
-	font-style: var(--_ui5_input_error_warning_font_style);
+	font-style: var(--_ui5_textarea_error_warning_font_style);
+	font-weight: var(--_ui5_textarea_error_warning_font_weight);
+	text-indent: var(--_ui5_textarea_error_warning_text_indent);
+	padding-top: var(--_ui5_textarea_padding_top_error_warning);
+	padding-bottom: var(--_ui5_textarea_padding_bottom_error_warning);
 }
 
-:host([value-state="Warning"]) .ui5-textarea-inner {
-	font-weight: var(--_ui5_input_warning_font_weight);
-}
-
-:host([value-state="Warning"]:not([readonly])) .ui5-textarea-inner {
-	background: var(--sapField_WarningBackgroundStyle);
-	background-color: var(--sapField_WarningBackground);
-}
-
-:host([value-state="Warning"]:not([readonly]):not([focused]):hover) .ui5-textarea-inner {
-	box-shadow: var(--sapContent_Critical_Shadow);
-	background-color: var(--sapField_Hover_Background);
+:host([value-state="Error"]:not([readonly]):not([disabled])) .ui5-textarea-wrapper,
+:host([value-state="Warning"]:not([readonly]):not([disabled])) .ui5-textarea-wrapper {
+	border-style: var(--_ui5_textarea_error_warning_border_style);
+	border-width: var(--_ui5_textarea_state_border_width);
 }
 
-:host([value-state="Warning"][focused]:not([readonly])) .ui5-textarea-inner {
-	background-image: none;
-	box-shadow: var(--_ui5_textarea_value_state_warning_focus_box_shadow);
-	background-color: var(--sapField_Focus_Background);
+:host([value-state="Error"]:not([readonly]):not([disabled])) .ui5-textarea-inner,
+:host([value-state="Warning"]:not([readonly]):not([disabled])) .ui5-textarea-inner,
+:host([value-state="Information"]:not([readonly]):not([disabled])) .ui5-textarea-inner {
+	border-style: none;
 }
 
-:host([value-state="Error"]:not([readonly])) .ui5-textarea-inner {
-	font-weight: var(--_ui5_input_error_font_weight);
-	background: var(--sapField_InvalidBackgroundStyle);
-	background-color: var(--sapField_InvalidBackground);
+:host([value-state="Warning"]:not([readonly]):not([disabled])) .ui5-textarea-wrapper {
+	background: var(--sapField_WarningBackgroundStyle);
+	background-color: var(--sapField_WarningBackground);
+	border-color: var(--_ui5_textarea_value_state_warning_border_color);
```

Comment:
border-color: var(--_ui5_textarea_value_state_warning_border_color); do we need a parameter, it is used only here with value var(--sapField_WarningColor); for all supported themes